### PR TITLE
Make sidecar provider configurable

### LINKED ISSUE #7997: [stable/grafana] Using sidecar and dashboardProvider causes problems
**Is this a request for help?**:
No

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG

**Version of Helm and Kubernetes**:
Kubernetes 1.9.7
Helm 2.10.0

**Which chart**:
stable/grafana:1.14.8

**What happened**:
When both dashboardProviders with a provider named "default" (example for values file) and sidecar dashboard import are enabled, all dashboards are constantly imported and deleted. Both providers feel responsible for all imported dashboards and delete the dashboards imported by the other provider, since it can't find the file locally.

**What you expected to happen**:
Dashboards of both providers should be installed.

**How to reproduce it** (as minimally and precisely as possible):
values.yaml:
```
dashboardProviders:
  dashboardproviders.yaml:
    apiVersion: 1
    providers:
    - name: 'default'
      orgId: 1
      folder: ''
      type: file
      disableDeletion: false
      editable: true
      options:
        path: /var/lib/grafana/dashboards/default
dashboards:
  default:
    kubernetes-cluster-monitoring:
      datasource: Prometheus
      gnetId: 315
      revision: 3
sidecar:
  dashboards:
    enabled: true
```
**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by pulledtim**
#### What this PR does / why we need it:
Give the user the possibility to configure the dashboard provider in the sidecar importer.

#### Which issue this PR fixes
    - fixes #7997

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/grafana/templates/configmap-dashboard-provider.yaml [hunk 1]
```diff
@@ -16,11 +16,11 @@ data:
   provider.yaml: |-
     apiVersion: 1
     providers:
-    - name: 'default'
-      orgId: 1
-      folder: ''
-      type: file
-      disableDeletion: false
+    - name: '{{ .Values.sidecar.dashboards.provider.name }}'
```
Comment (cpanato) [inline stable/grafana/templates/configmap-dashboard-provider.yaml:19]: can you please document these in the values.yaml?

Comment (pulledtim) [reply stable/grafana/templates/configmap-dashboard-provider.yaml:19]: They already are included in the values.yaml and documented there. Did you miss them or are they not sufficient, in which case I gladly extend them.

File: stable/grafana/values.yaml [hunk 2]
```diff
@@ -239,24 +239,42 @@ smtp:
...
   dashboards:
+    # enabled dashboard import
     enabled: false
     # label that the configmaps with dashboards are marked with
     label: grafana_dashboard
     # folder in the pod that should hold the collected dashboards
     folder: /tmp/dashboards
+    # provider configuration that lets grafana manage the dashboards
+    provider:
+      # name of the provider, should be unique
```
Comment (pulledtim) [inline stable/grafana/values.yaml:278]: @cpanato Regarding your remark

File: stable/grafana/templates/configmap-dashboard-provider.yaml (hunk 1 above)
Comment (cpanato) [reply stable/grafana/templates/configmap-dashboard-provider.yaml:19]: i'm sorry, i think one and wrote another :(. I meat to document in the readme.md

**Comment by sta-szek** [comment]
@Tim-Smyth any update here? i see conflicts

**Pull request closed without merging**